### ISSUE TITLE
Support for Reolink Camera's

### ISSUE THREAD

**Issue by BartDeWorm** [AdyRock/com.onvif#35]
Hi,
I bought a Reolink NVR and some Reolink Camera's; added them to Homey.
But fetch the image doesn't work.

In the logging I see:
http://192.168.68.139:80/cgi-bin/api.cgi?cmd=onvifSnapPic&amp;channel=0
http://192.168.68.139:80/cgi-bin/api.cgi?cmd=onvifSnapPic&channel=1

But on the reolink site the string must be like:
http://(ip address)/cgi-bin/api.cgi?cmd=Snap&channel=0&rs=(any combination of numbers and letters)&user=(user name)&password=(user password)

https://support.reolink.com/hc/en-us/articles/360007011233-How-to-Capture-Live-JPEG-Image-of-Reolink-Cameras-via-Web-Browsers

Is it possible to change this in any way?
Because when I test the link of Reolink it works fine.

Thx in advance!

Full log below:
Log ID: 2022-11-10T08:13:39.231Z
App version 2.0.11

8:13:39.231: 
_request (192.168.68.139): {
  "hostname": "192.168.68.139",
  "port": 8000,
  "path": "/onvif/media_service",
  "agent": false,
  "timeout": 15000,
  "headers": {
    "Content-Type": "application/soap+xml",
    "Content-Length": 1069,
    "charset": "utf-8"
  },
  "method": "POST"
}
Body: <s:Envelope xmlns:s="http://www.w3.org/2003/05/soap-envelope" xmlns:a="http://www.w3.org/2005/08/addressing"><s:Header><a:MessageID> urn:uuid:6ace50db-ab36-76ec-d803-3393cfc06626</a:MessageID><Security s:mustUnderstand="1" xmlns="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd"><UsernameToken><Username>admin</Username><Password Type="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-username-token-profile-1.0#PasswordDigest">PAspjsN9bjvckT6NrQ3VaAzbAdk=</Password><Nonce EncodingType="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-soap-message-security-1.0#Base64Binary">rHVyfMK79/q+IT1YMPhRtQ==</Nonce><Created xmlns="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd">2022-11-10T08:13:39.152Z</Created></UsernameToken></Security></s:Header><s:Body xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"><GetSnapshotUri xmlns="http://www.onvif.org/ver10/media/wsdl"><ProfileToken>000</ProfileToken></GetSnapshotUri></s:Body></s:Envelope>

8:13:39.568: 
_request response (192.168.68.139): <?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://www.w3.org/2003/05/soap-envelope" xmlns:SOAP-ENC="http://www.w3.org/2003/05/soap-encoding" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:wsa="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:wsdd="http://schemas.xmlsoap.org/ws/2005/04/discovery" xmlns:wsa5="http://www.w3.org/2005/08/addressing" xmlns:xmime="http://tempuri.org/xmime.xsd" xmlns:xmime5="http://www.w3.org/2005/05/xmlmime" xmlns:xop="http://www.w3.org/2004/08/xop/include" xmlns:wsrfbf="http://docs.oasis-open.org/wsrf/bf-2" xmlns:tt="http://www.onvif.org/ver10/schema" xmlns:wstop="http://docs.oasis-open.org/wsn/t-1" xmlns:wsrfr="http://docs.oasis-open.org/wsrf/r-2" xmlns:ns1="http://www.onvif.org/ver10/actionengine/wsdl" xmlns:tev="http://www.onvif.org/ver10/events/wsdl" xmlns:ns10="http://www.onvif.org/ver10/events/wsdl/PullPointBinding" xmlns:ns11="http://www.onvif.org/ver10/events/wsdl/CreatePullPointBinding" xmlns:ns12="http://www.onvif.org/ver10/events/wsdl/PausableSubscriptionManagerBinding" xmlns:ns13="http://www.onvif.org/ver10/network/wsdl/RemoteDiscoveryBinding" xmlns:ns14="http://www.onvif.org/ver10/network/wsdl/DiscoveryLookupBinding" xmlns:tdn="http://www.onvif.org/ver10/network/wsdl" xmlns:ns3="http://www.onvif.org/ver20/analytics/wsdl/RuleEngineBinding" xmlns:ns4="http://www.onvif.org/ver20/analytics/wsdl/AnalyticsEngineBinding" xmlns:tan="http://www.onvif.org/ver20/analytics/wsdl" xmlns:ns5="http://www.onvif.org/ver10/events/wsdl/PullPointSubscriptionBinding" xmlns:ns6="http://www.onvif.org/ver10/events/wsdl/EventBinding" xmlns:ns7="http://www.onvif.org/ver10/events/wsdl/SubscriptionManagerBinding" xmlns:ns8="http://www.onvif.org/ver10/events/wsdl/NotificationProducerBinding" xmlns:wsnt="http://docs.oasis-open.org/wsn/b-2" xmlns:ns9="http://www.onvif.org/ver10/events/wsdl/NotificationConsumerBinding" xmlns:tad="http://www.onvif.org/ver10/analyticsdevice/wsdl" xmlns:tds="http://www.onvif.org/ver10/device/wsdl" xmlns:timg="http://www.onvif.org/ver20/imaging/wsdl" xmlns:tls="http://www.onvif.org/ver10/display/wsdl" xmlns:tmd="http://www.onvif.org/ver10/deviceIO/wsdl" xmlns:tptz="http://www.onvif.org/ver20/ptz/wsdl" xmlns:trc="http://www.onvif.org/ver10/recording/wsdl" xmlns:trp="http://www.onvif.org/ver10/replay/wsdl" xmlns:trt="http://www.onvif.org/ver10/media/wsdl" xmlns:trv="http://www.onvif.org/ver10/receiver/wsdl" xmlns:tse="http://www.onvif.org/ver10/search/wsdl" xmlns:ter="http://www.onvif.org/ver10/error" xmlns:tns1="http://www.onvif.org/ver10/topics" xmlns:wsse="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd" xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd"><SOAP-ENV:Header><wsa5:MessageID> urn:uuid:6ace50db-ab36-76ec-d803-3393cfc06626</wsa5:MessageID></SOAP-ENV:Header><SOAP-ENV:Body><trt:GetSnapshotUriResponse><trt:MediaUri><tt:Uri>[http://192.168.68.139:80/cgi-bin/api.cgi?cmd=onvifSnapPic&amp;channel=0](http://192.168.68.139/cgi-bin/api.cgi?cmd=onvifSnapPic&amp;channel=0)</tt:Uri><tt:InvalidAfterConnect>true</tt:InvalidAfterConnect><tt:InvalidAfterReboot>true</tt:InvalidAfterReboot><tt:Timeout>PT10S</tt:Timeout></trt:MediaUri></trt:GetSnapshotUriResponse></SOAP-ENV:Body></SOAP-ENV:Envelope>

8:13:39.588: Fetching (192.168.68.139 (Ch1)) NOW image with Digest Auth. From: [http://192.168.68.139:80/cgi-bin/api.cgi?cmd=onvifSnapPic&channel=1](http://192.168.68.139/cgi-bin/api.cgi?cmd=onvifSnapPic&channel=1)
8:13:39.750: SnapShot fetch result (192.168.68.139 (Ch1)): Status: true, Message: OK, Code: 200

8:13:39.767: Now Image Updated (192.168.68.139 (Ch1))
8:13:41.301: 
_request (192.168.68.139): {
  "hostname": "192.168.68.139",
  "port": 8000,
  "path": "/onvif/media_service",
  "agent": false,
  "timeout": 15000,
  "headers": {
    "Content-Type": "application/soap+xml",
    "Content-Length": 1069,
    "charset": "utf-8"
  },
  "method": "POST"
}
Body: <s:Envelope xmlns:s="http://www.w3.org/2003/05/soap-envelope" xmlns:a="http://www.w3.org/2005/08/addressing"><s:Header><a:MessageID> urn:uuid:30c2cb37-57bd-5721-b87d-dd615c8ac8d3</a:MessageID><Security s:mustUnderstand="1" xmlns="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd"><UsernameToken><Username>admin</Username><Password Type="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-username-token-profile-1.0#PasswordDigest">Y1rhwwDrVNz4i5e2beJD1/IwxHc=</Password><Nonce EncodingType="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-soap-message-security-1.0#Base64Binary">77dIY3IpRBM+zM1utZ22pw==</Nonce><Created xmlns="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd">2022-11-10T08:13:41.222Z</Created></UsernameToken></Security></s:Header><s:Body xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"><GetSnapshotUri xmlns="http://www.onvif.org/ver10/media/wsdl"><ProfileToken>000</ProfileToken></GetSnapshotUri></s:Body></s:Envelope>

8:13:41.447: 
_request response (192.168.68.139): <?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://www.w3.org/2003/05/soap-envelope" xmlns:SOAP-ENC="http://www.w3.org/2003/05/soap-encoding" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:wsa="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:wsdd="http://schemas.xmlsoap.org/ws/2005/04/discovery" xmlns:wsa5="http://www.w3.org/2005/08/addressing" xmlns:xmime="http://tempuri.org/xmime.xsd" xmlns:xmime5="http://www.w3.org/2005/05/xmlmime" xmlns:xop="http://www.w3.org/2004/08/xop/include" xmlns:wsrfbf="http://docs.oasis-open.org/wsrf/bf-2" xmlns:tt="http://www.onvif.org/ver10/schema" xmlns:wstop="http://docs.oasis-open.org/wsn/t-1" xmlns:wsrfr="http://docs.oasis-open.org/wsrf/r-2" xmlns:ns1="http://www.onvif.org/ver10/actionengine/wsdl" xmlns:tev="http://www.onvif.org/ver10/events/wsdl" xmlns:ns10="http://www.onvif.org/ver10/events/wsdl/PullPointBinding" xmlns:ns11="http://www.onvif.org/ver10/events/wsdl/CreatePullPointBinding" xmlns:ns12="http://www.onvif.org/ver10/events/wsdl/PausableSubscriptionManagerBinding" xmlns:ns13="http://www.onvif.org/ver10/network/wsdl/RemoteDiscoveryBinding" xmlns:ns14="http://www.onvif.org/ver10/network/wsdl/DiscoveryLookupBinding" xmlns:tdn="http://www.onvif.org/ver10/network/wsdl" xmlns:ns3="http://www.onvif.org/ver20/analytics/wsdl/RuleEngineBinding" xmlns:ns4="http://www.onvif.org/ver20/analytics/wsdl/AnalyticsEngineBinding" xmlns:tan="http://www.onvif.org/ver20/analytics/wsdl" xmlns:ns5="http://www.onvif.org/ver10/events/wsdl/PullPointSubscriptionBinding" xmlns:ns6="http://www.onvif.org/ver10/events/wsdl/EventBinding" xmlns:ns7="http://www.onvif.org/ver10/events/wsdl/SubscriptionManagerBinding" xmlns:ns8="http://www.onvif.org/ver10/events/wsdl/NotificationProducerBinding" xmlns:wsnt="http://docs.oasis-open.org/wsn/b-2" xmlns:ns9="http://www.onvif.org/ver10/events/wsdl/NotificationConsumerBinding" xmlns:tad="http://www.onvif.org/ver10/analyticsdevice/wsdl" xmlns:tds="http://www.onvif.org/ver10/device/wsdl" xmlns:timg="http://www.onvif.org/ver20/imaging/wsdl" xmlns:tls="http://www.onvif.org/ver10/display/wsdl" xmlns:tmd="http://www.onvif.org/ver10/deviceIO/wsdl" xmlns:tptz="http://www.onvif.org/ver20/ptz/wsdl" xmlns:trc="http://www.onvif.org/ver10/recording/wsdl" xmlns:trp="http://www.onvif.org/ver10/replay/wsdl" xmlns:trt="http://www.onvif.org/ver10/media/wsdl" xmlns:trv="http://www.onvif.org/ver10/receiver/wsdl" xmlns:tse="http://www.onvif.org/ver10/search/wsdl" xmlns:ter="http://www.onvif.org/ver10/error" xmlns:tns1="http://www.onvif.org/ver10/topics" xmlns:wsse="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd" xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd"><SOAP-ENV:Header><wsa5:MessageID> urn:uuid:30c2cb37-57bd-5721-b87d-dd615c8ac8d3</wsa5:MessageID></SOAP-ENV:Header><SOAP-ENV:Body><trt:GetSnapshotUriResponse><trt:MediaUri><tt:Uri>[http://192.168.68.139:80/cgi-bin/api.cgi?cmd=onvifSnapPic&amp;channel=0](http://192.168.68.139/cgi-bin/api.cgi?cmd=onvifSnapPic&amp;channel=0)</tt:Uri><tt:InvalidAfterConnect>true</tt:InvalidAfterConnect><tt:InvalidAfterReboot>true</tt:InvalidAfterReboot><tt:Timeout>PT10S</tt:Timeout></trt:MediaUri></trt:GetSnapshotUriResponse></SOAP-ENV:Body></SOAP-ENV:Envelope>

8:13:41.478: Fetching (192.168.68.139 (Ch1)) NOW image with Digest Auth. From: [http://192.168.68.139:80/cgi-bin/api.cgi?cmd=onvifSnapPic&channel=1](http://192.168.68.139/cgi-bin/api.cgi?cmd=onvifSnapPic&channel=1)
8:13:41.648: SnapShot fetch result (192.168.68.139 (Ch1)): Status: true, Message: OK, Code: 200

8:13:41.663: Now Image Updated (192.168.68.139 (Ch1))

**Comment by BartDeWorm** [comment]
Never mind! It works now!
Typo in the User Snapshot URL and Channel must be on 0 or -1.

Thx Bart

**Comment by BartDeWorm** [comment]
Only the motion part doesn't work.
I'm reading this issue but I can't get it..
https://github.com/AdyRock/com.onvif/issues/10

Any help?

**Comment by AdyRock** [comment]
Glad the image is now working. 
The key part of issue #10 is basically making sure that motion detection is enabled in the app, e.g. the button is on 
![Screenshot_20221110_093851_Homey](https://user-images.githubusercontent.com/38184396/201055378-daf9ee99-9f0a-4b9d-b428-214b3b31457e.jpg)

**Comment by BartDeWorm** [comment]
Hmmm this button doesn't lights up automatically:
![afbeelding](https://user-images.githubusercontent.com/50958122/201060573-3eb1bc65-724a-47b2-aac5-76a594fca10d.png)

On the screen of my NVR I see there is motion detected.. but still it doesn't work.
I would like to receive a snapshot when there is motion on the camera.

I played also with the 'Bewegings trigger > Product Indeling (sensor/camera). But nothing happens.

But now it works! I changed 'Bewegings trigger' > 'Voorkeur voor Pull Events' from No to Yes.
